apps/meteor/app/api/server/v1/chat.ts

```diff
@@ -492,33 +492,51 @@ API.v1.addRoute(
 );
 
 API.v1.addRoute(
-	'chat.getPinnedMessages',
-	{ authRequired: true, validateParams: isChatGetPinnedMessagesProps },
-	{
-		async get() {
-			const { roomId } = this.queryParams;
-			const { offset, count } = await getPaginationItems(this.queryParams);
+  'chat.getPinnedMessages',
+  { authRequired: true, validateParams: isChatGetPinnedMessagesProps },
+  {
+    async get() {
+      const { roomId, sort } = this.queryParams;
+      const { offset, count } = await getPaginationItems(this.queryParams);
+
+      if (!(await canAccessRoomIdAsync(roomId, this.userId))) {
+        throw new Meteor.Error('error-not-allowed', 'Not allowed');
+      }
+
+      let sortParam: Record<string, 1 | -1> = {};  // Inicializa o parâmetro de ordenação
+
+      // Agora, a ordenação está sendo aplicada
+      try {
+        if (sort) {
+          sortParam = JSON.parse(sort);  // Converte o parâmetro 'sort' da string JSON para um objeto
```

Comment:
just a suggestion - I noticed you're using manual JSON.parse() for the sort parameter. The codebase has a built-in parseJsonQuery() method that handles this securely, maybe it could be used here too?